File: uikit/src/main/java/com/cometchat/pro/uikit/Sticker/model/Sticker.java
package com.cometchat.pro.uikit.Sticker.model;

import android.os.Parcel;
import android.os.Parcelable;

public class Sticker implements Parcelable {
    String name;
    String url;
    String setName;

    public Sticker(String name,String url,String setName) {
        this.name = name;
        this.url = url;
        this.setName = setName;
    }

    protected Sticker(Parcel in) {
        name = in.readString();
        url = in.readString();
        setName = in.readString();
    }

    public static final Creator<Sticker> CREATOR = new Creator<Sticker>() {
        @Override
        public Sticker createFromParcel(Parcel in) {
            return new Sticker(in);
        }

        @Override
        public Sticker[] newArray(int size) {
            return new Sticker[size];
        }
    };

    public String getName() {
        return name;
    }

    public String getUrl() {
        return url;
    }

    public String getSetName() {
        return setName;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(name);
        parcel.writeString(url);
        parcel.writeString(setName);
    }
}

File: uikit-kotlin/src/main/java/com/cometchat/pro/uikit/sticker/model/Sticker.kt
package com.cometchat.pro.uikit.sticker.model

import android.os.Parcel
import android.os.Parcelable

class Sticker : Parcelable{
    var name: String? = null
    var url: String? = null
    var setName: String? = null


    constructor(parcel: Parcel) {
        name = parcel.readString()
        url = parcel.readString()
        setName = parcel.readString()
    }

    constructor(name: String?, url: String?, setName: String?) {
        this.name = name
        this.url = url
        this.setName = setName
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(name)
        parcel.writeString(url)
        parcel.writeString(setName)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Sticker> {
        override fun createFromParcel(parcel: Parcel): Sticker {
            return Sticker(parcel)
        }

        override fun newArray(size: Int): Array<Sticker?> {
            return arrayOfNulls(size)
        }
    }

}